### PR TITLE
School Library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by jugosack**
### In this PR, I added 3 classes.

**Features:**

- Setup the linter configuration
- Added the person.rb
- Added the student.rb
- Added the teacher.rb
- Added README.md file

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age, :parent_permission
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (Gambit142) [inline person.rb]: - [ ] The requirements state that the `of_age` method should be private. Currently, it is a public method. Please make it a private method.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require './person'
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(age, classroom, name: 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
```
Comment (Gambit142) [inline student.rb]: - [ ] When I try to create an instance of the `Student` class, I get this error:

image

When initializing this class, the `age` and `classroom` data are required/default parameters while the `name` and `parent_permission` are keyword parameters with default values based on the way you initialized the class. Therefore, in your super method, you need to specify the keyword parameters there:
`super(age, name: name, parent_permission: parent_permission)`. 
When you are creating a new instance, you need to pass in the default arguments first before the key arguments:

`student = Student.new(15, 'Math', name: 'James', parent_permission: true)`

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require './person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age, specialization, name: 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @specialization = specialization
+  end
```
Comment (Gambit142) [inline teacher.rb]: - [ ] When I try to create an instance of the `Teacher` class, I get this error:

image

When initializing this class, the `age` and `specialization` data are required/default parameters while the `name` and `parent_permission` are keyword parameters with default values based on the way you initialized the class. Therefore, in your `super` method, you need to specify the keyword parameters there:
`super(age, name: name, parent_permission: parent_permission)`. 
When you are creating a new instance, you need to pass in the default arguments first before the key arguments:

`teacher = Teacher.new(15, 'Math', name: 'James', parent_permission: true)`

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @jugosack 👋, 

Good job so far!

## Highlights
- Rubocop linter check passed ✔️ 
- Created a `Person` class ✔️ 
- Created a `Teacher` class ✔️ 
- Created a `Student` class ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: person.rb (hunk 1 above)
Comment (danmainah) [reply person.rb]: - [ ] @jugosack great work implementing the changes requested however you have made`can_use_services?` method private and its suppose to be public.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @jugosack , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by pinkmoon25 — APPROVED** [review]
Hi @jugosack , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Linters :heavy_check_mark: 
Gitflow :heavy_check_mark: 
Readme :heavy_check_mark:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**